File: Examples/0b/ProbB.vb
Option Explicit On

Module HelloWorld

	' The Main subroutine is the entry point for your program;
	' make sure to include it! This file can be used as a
	' template for solving all Code Quest problems
    Sub Main(args As String())
		' All inputs starts with the number of test cases.
		' These next two lines should appear in every solution.
        Dim i As Integer
        Int32.TryParse(Console.ReadLine(), i)
		' Now we need to do something for each test case. This
		' For loop allows us to do that. These two lines are the
		' last two common to all problems; inside the loop, it'll
		' be different for each problem.
        Dim test As Integer = 0
        For test = 0 To (i - 1)
			' For Not So Self-Driving, we need to read in our data
			Dim dataString As String
			dataString = Console.ReadLine()
			' Then pull out the numbers to work on
			Dim data As String()
			data = dataString.Split(New Char(){":"c})
			Dim velocity As Double
			Dim distance As Double
			Double.TryParse(data(0), velocity)
			Double.TryParse(data(1), distance)
			' Calculate time to collision
			If velocity = 0.0 Then
				Console.WriteLine("SAFE")
			Else
				Dim time As Double
				time = distance / velocity
				' Respond accordingly
				If time <= 1.0 Then
					Console.WriteLine("SWERVE")
				Else If time <= 5.0 Then
					Console.WriteLine("BRAKE")
				Else
					Console.WriteLine("SAFE")
				End If
			End If
        Next
    End Sub

End Module

File: Examples/0b/ProbB.cs
/*
 * ProbA.cs
 *
 * Copyright (c) 2020 Lockheed Martin (LM) as an unpublished work. 
 * All Rights Reserved.
 *
 * This file does not contain any Lockheed Martin Proprietary Information.
 */
using System;

/**
 * Class CodeQuest
 * 
 * This class is used to run the Not So Self-Driving test problem.
 * You are strongly encouraged to use this as a template
 * for use in solving other problems.
 */
class CodeQuest {
	// This is the Main method, which is the entry 
	// point for your program. Without this, your
	// program can't be run.
	static void Main(string[] args) {
		// This line reads the first line of the input and
		// converts it to an integer. All inputs start with
		// the number of test cases, so this line should be
		// at the beginning of every solution you write.
		int numTestCases = Convert.ToInt32(Console.ReadLine());
		
		// You'll want to do something for each test case.
		// This for loop controls that behavior. This line
		// is the last one that will be common to all problems.
		for(int testCase = 0; testCase < numTestCases; testCase = testCase + 1){
			// Inside the for loop, we need to read the input line
			string text = Console.ReadLine();
			// then split out the separate numbers
			string[] data = text.Split(new Char[]{':'});
			
			// convert them to actual numbers
			double velocity = Convert.ToDouble(data[0]);
			double distance = Convert.ToDouble(data[1]);
			
			double time = distance / velocity;
			
			if(time <= 1.0){
				Console.WriteLine("SWERVE");
			}
			else if(time <= 5.0){
				Console.WriteLine("BRAKE");
			}
			else{
				Console.WriteLine("SAFE");
			}
		}
	}
}